pallets/creditcoin/src/helpers/external_address.rs

```diff
@@ -1,14 +1,63 @@
 use crate::{Blockchain, ExternalAddress};
 use base58::FromBase58;
+use core::convert::TryFrom;
+use frame_support::BoundedVec;
+use sp_core::ecdsa::Public;
+use sp_io::hashing::keccak_256;
 use sp_io::hashing::sha2_256;
 
+pub fn external_address_generator(
+	blockchain: &Blockchain,
+	reference: &ExternalAddress,
+	pkey: Public,
+) -> Option<ExternalAddress> {
+	match blockchain {
+		Blockchain::Luniverse | Blockchain::Ethereum | Blockchain::Rinkeby
+			if Etherlike::is_address(reference).is_some() =>
+		{
+			Some(Etherlike::from_public(&pkey))
+		},
+		Blockchain::Bitcoin => None,
+		Blockchain::Other(_) => None,
+		_ => None,
+	}
+}
+
+pub trait PublicToAddress {
+	type AddressType;
+	fn is_address(addr: &ExternalAddress) -> Option<Self::AddressType>;
+	fn from_public(pkey: &Public) -> ExternalAddress;
+}
+
+pub struct Etherlike;
```

Comment:
This should be `EtherLike`, or better would be a more descriptive name though I suck at naming so am failing to come up with a good alternative. Some thoughts EthStyleAddress or EVMAddress

Reply:
To my understanding it is a single word. Googled Headlike as an example. Same with Ethstyle, no? I like EVMAddress.

Reply:
Interesting, i just looked up fishlike (don't ask why i picked that word) and I guess you can just add "like" to any word, thanks for the english lesson lol.